### DISCUSSION TITLE
App runs but window does not appear and app does not loop

### DISCUSSION THREAD

**Discussion by alice-i-cecile** [bevyengine/bevy#2878] [Q&A] [answered]
I've been working on a [template repo](https://github.com/Leafwing-Studios/template-repo) for Bevy games and libraries.

The file structure is slightly different than the typical: both a `main.rs` and `lib.rs` exist within the `src` file, and `main` is recorded as `bin` in `Cargo.toml`.

When I call `cargo run`, my app runs "successfully", printing `Hello, World!". 
However:
- no main window appears
- the app does not loop: each main stage runs exactly once
- no errors, warnings or other diagnostics appear

This is isolated to this project, other simple demos and existing projects work totally fine.

I suspect the project structure is somehow interacting badly with the default winit-powered runner, but I have no idea how or why. Any thoughts would be welcome!

**Comment by DJMcNab** [comment, marked as answer]
You don't have the `winit` feature enabled in your bevy dependency, so the winit runner isn't on the default plugins

This is expected behaviour

As a side note, you need to add the `WindowDescriptor` before the `DefaultPlugins`, because otherwise it won't be registered properly.

**Reply by alice-i-cecile** [reply]
Aha! Curse my split design. Let me test this out.

**Reply by alice-i-cecile** [reply]
That worked! Thanks; sometimes it's incredibly hard to spot the obvious stuff when you've been staring at a project all day.
